src/rust/ide/src/double_representation/module.rs

```diff
@@ -598,18 +608,27 @@ pub fn locate
 ///
 /// The module is assumed to be in the file identified by the `method.file` (for the purpose of
 /// desugaring implicit extensions methods for modules).
+///
+/// The `name` parameter is the name of the module that contains `ast`. It affects how the `here`
+/// keyword is resolved.
 pub fn lookup_method
```

Comment:
At this point when I read this module I wonder what decides when a function is a method of `Info` and when it isn't? Perhaps most of them could go inside Info, and even Info could easily keep module name next to ast?

That only under your consideration, it is not blocker for merging PR.

Reply:
Leaving as-is, should be refactored in the future.

Functions being free and not in `Info` is mostly consequence of `Info` being introduced later than them. We don't want `Info` to contain name, as most of the methods don't need it.  
Another issue is the ergonomy of usage — having to wrap Ast into info every single time is not convenient. Some places also don't know name but want to process Ast.

The lookup function however should not really be a method should not be a method of plain info but something that contains both ast and name.  If other functions working on such name+ast pair appear, we should consider introducing it as a new structure (NamedInfo or sth similar). Info should not be structurally changed.